[WEBSITE-219] - Properly navigate to dependencies

michaelmccaskill opened the pull request:
Related to issue # [WEBSITE-219]

Summary of this pull request: 

The plugin being displayed is in the redux state. This is loaded when
the component is mounted. The plugin to load is read via the router
parameter `pluginName`. But nothing was monitoring subsequent changes
to said parameter. The solution is to implement
`componentWillReceiveProps` and determine if the parameter has indeed
changed, and if so get the new plugin. This elegantly handles forward
and back navigation.

rtyler commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+   # Navigate to the newly cloned directory
+   cd plugin-site
+   # Assign the original repo to a remote called "upstream"
+   git remote add upstream https://github.com/jenkins-infra /plugin-site.git
```
There's a space between `jenkins-infra` and `/plugin-site.git` FWIW

rtyler commented:
The "Files changed" view is really difficult to parse through with all the deleted diffs. Would you mind rebasing this against the current master now that #1 has been merged?

Commit:
[WEBSITE-219] - Properly navigate to dependencies

The plugin being displayed is in the redux state. This is loaded when
the component is mounted. The plugin to load is read via the router
parameter `pluginName`. But nothing was monitoring subsequent changes
to said parameter. The solution is to implement
`componentWillReceiveProps` and determine if the parameter has indeed
changed, and if so get the new plugin. This elegantly handles forward
and back navigation.

rtyler commented:
@michaelmccaskill I didn't notice this when @kohsuke originally imported the tree, but the `Jenkinsfile` won't properly run on [ci.jenkins.io](https://ci.jenkins.io) since it has an empty labeled `node` declaration.

I think just changing to `node('docker') {` will suffice, more information on our labels can be [found here](https://github.com/jenkins-infra/documentation/blob/master/ci.adoc)

michaelmccaskill commented:
Oh ok. Thanks for the information. I wasn't aware of these was required. I will make the change.

rtyler commented:
@michaelmccaskill Yeah, it has only been recently documented, my intent is that some of this kind of things finds its way into @daniel-beck's developer documentation at some point. 

How to use our own Jenkins is...not well documented right now past that doc :frowning_man:

Commit:
[WEBSITE-219] - Use docker node

Per Tyler we need to specify the node to ensure docker is available.
